service/frontend/workflow_handler.go

```diff
@@ -4351,26 +4351,64 @@ func (wh *WorkflowHandler) validateTaskQueue(t *taskqueuepb.TaskQueue) error {
 	return nil
 }
 
-func (wh *WorkflowHandler) validateBuildIdOrderingUpdate(
-	req *workflowservice.UpdateWorkerBuildIdOrderingRequest,
+func (wh *WorkflowHandler) validateBuildIdCompatabilityUpdate(
+	req *workflowservice.UpdateWorkerBuildIdCompatabilityRequest,
 ) error {
-	errstr := "request to update worker build id ordering requires:"
+	errstr := "request to update worker build id compatability requires:"
 	hadErr := false
+
+	checkIdLen := func(id string) {
+		if len(id) > wh.config.WorkerBuildIdSizeLimit() {
+			errstr += fmt.Sprintf(" Worker build IDs to be no larger than %v characters",
```

Comment:
maybe collect errors in a `[]string` and join them at the end? then you don't need hadErr

Reply:
Good point